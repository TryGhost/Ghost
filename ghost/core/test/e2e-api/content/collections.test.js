const {agentProvider, fixtureManager, matchers} = require('../../utils/e2e-framework');
const {anyISODateTime, anyObjectId} = matchers;

const collectionMatcher = {
    id: anyObjectId,
    created_at: anyISODateTime,
    updated_at: anyISODateTime
};

describe('Collections Content API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getContentAPIAgent();
        await fixtureManager.init('users', 'api_keys');
        await agent.authenticate();
    });

    it('Can request a collection by slug and id', async function () {
        const {body: {collections: [collection]}} = await agent
            .get(`collections/slug/featured`)
            .expectStatus(200)
            .matchBodySnapshot({
                collections: [collectionMatcher]
            });

        await agent
            .get(`collections/${collection.id}`)
            .expectStatus(200)
            .matchBodySnapshot({
                collections: [collectionMatcher]
            });
    });
});
