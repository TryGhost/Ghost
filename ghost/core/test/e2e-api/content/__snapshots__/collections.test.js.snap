// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collections Content API Can request a collection by slug and id 1: [body] 1`] = `
Object {
  "collections": Array [
    Object {
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "description": "Featured posts",
      "feature_image": null,
      "filter": "featured:true",
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "slug": "featured",
      "title": "Featured",
      "type": "automatic",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
    },
  ],
}
`;

exports[`Collections Content API Can request a collection by slug and id 2: [body] 1`] = `
Object {
  "collections": Array [
    Object {
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "description": "Featured posts",
      "feature_image": null,
      "filter": "featured:true",
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "slug": "featured",
      "title": "Featured",
      "type": "automatic",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
    },
  ],
}
`;
